pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -856,6 +886,61 @@ func (r *ReconcileClusterDeployment) setupClusterInstallServiceAccount(namespace
 	return currentSA, nil
 }
 
+func (r *ReconcileClusterDeployment) selectorPodWatchHandler(a handler.MapObject) []reconcile.Request {
+	retval := []reconcile.Request{}
+
+	pod := a.Object.(*corev1.Pod)
+	if pod == nil {
+		// Wasn't a Pod, bail out. This should not happen.
+		log.Errorf("Error converting MapObject.Object to Pod. Value: %+v", a.Object)
+		return retval
+	}
+	if pod.Labels != nil {
+		if cdName, ok := pod.Labels[install.ClusterDeploymentNameLabel]; ok {
+			if cdNamespace, ok := pod.Labels[install.ClusterDeploymentNamespaceLabel]; ok {
+				retval = append(retval, reconcile.Request{NamespacedName: types.NamespacedName{
+					Name:      cdName,
+					Namespace: cdNamespace,
+				}})
+			}
+		}
+	}
+	return retval
+}
+
+func (r *ReconcileClusterDeployment) updateInstallRetriesTotalMetric(clusterDeploymentName string, clusterDeploymentNamespace string, cdLog log.FieldLogger) {
+	labels := map[string]string{"cluster_deployment": clusterDeploymentName, "namespace": "true"}
+	installerPodLabels := map[string]string{install.ClusterDeploymentNameLabel: clusterDeploymentName, install.ClusterDeploymentNamespaceLabel: clusterDeploymentNamespace, install.InstallJobLabel: "true"}
+	currentRetiesMetric, err := metricClusterDeploymentInstallRetriesTotal.GetMetricWith(labels)
+	if err != nil {
+		cdLog.Error("Error getting cluster deployment install retries total metrics.")
+	} else {
+		cdLog.Infof("CURRENT VALUE:%v", currentRetiesMetric)
+		pods := &corev1.PodList{}
+		err = r.Client.List(context.Background(), client.MatchingLabels(installerPodLabels), pods)
```

Comment:
Per below, lets drop the Namepace label, and instead of client.MatchingLabels just build your own client.ListOptions with both a label selector, and a namespace.

Reply:
This commend still needs to be addressed, you need to replace client.MatchingLables with a ListOptions that includes your labels, and the namespace. (as we dropped the namespace from the labels themselves)